mathics/core/formatter.py

```diff
@@ -119,3 +155,391 @@ def add_conversion_fn(cls, module_fn_name=None) -> None:
+            number_as_text = self.box_options.get("number_as_text", False)
+
+        def render(format, string):
+            encoded_text = encode_mathml(string)
+            return format % encoded_text
+
+        if text.startswith('"') and text.endswith('"'):
+            show_string_characters = options.get("show_string_characters", None)
+            if show_string_characters is None:
+                show_string_characters = (
+                    self.box_options["System`ShowStringCharacters"] is SymbolTrue
+                )
+
+            if show_string_characters:
+                return render("<ms>%s</ms>", text[1:-1])
+            else:
+                outtext = ""
+                for line in text[1:-1].split("\n"):
+                    outtext += render("<mtext>%s</mtext>", line)
+                return outtext
+        elif (
+            text
+            and not number_as_text
+            and ("0" <= text[0] <= "9" or text[0] in (".", "-"))
+        ):
+            return render("<mn>%s</mn>", text)
+        else:
+            if text in operators or text in extra_operators:
+                if text == "\u2146":
+                    return render(
+                        '<mo form="prefix" lspace="0.2em" rspace="0">%s</mo>', text
+                    )
+                if text == "\u2062":
+                    return render(
+                        '<mo form="prefix" lspace="0" rspace="0.2em">%s</mo>', text
+                    )
+                return render("<mo>%s</mo>", text)
+            elif is_symbol_name(text):
+                return render("<mi>%s</mi>", text)
+            else:
+                outtext = ""
+                for line in text.split("\n"):
+                    outtext += render("<mtext>%s</mtext>", line)
+                return outtext
+
+    def boxes_to_tex(self, **options) -> str:
+        text = self.value
+
+        def render(format, string, in_text=False):
+            return format % encode_tex(string, in_text)
+
+        if text.startswith('"') and text.endswith('"'):
+            show_string_characters = options.get("show_string_characters", None)
+            if show_string_characters is None:
+                show_string_characters = (
+                    self.box_options["System`ShowStringCharacters"] is SymbolTrue
+                )
+            # In WMA, ``TeXForm`` never adds quotes to
+            # strings, even if ``InputForm`` or ``FullForm``
+            # is required, to so get the standard WMA behaviour,
+            # this option is set to False:
+            # show_string_characters = False
+
+            if show_string_characters:
+                return render(r'\text{"%s"}', text[1:-1], in_text=True)
+            else:
+                return render(r"\text{%s}", text[1:-1], in_text=True)
+        elif text and text[0] in "0123456789-.":
+            return render("%s", text)
+        else:
+            # FIXME: this should be done in a better way.
```

Comment:
I believe we now have everything in place from the Mathics-Scanner repository to be able to do the single-character or operator entries in a table driven fashion. 

However this is still a bit wrong since it should be done when extracting text from an operator.